src/main/java/at/hannibal2/skyhanni/features/misc/GeorgeDisplay.kt

```diff
@@ -0,0 +1,113 @@
+    )
+
+    private var display = listOf<Renderable>()
+
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryOpenEvent) {
+        if (!isEnabled()) return
+        if (event.inventoryName != "Offer Pets") return
+        val stack = event.inventoryItems[41] ?: return
+        if (stack.cleanName() != "+1 Taming Level Cap") return
+        display = listBuilding(stack.getLore())
+    }
+
+    private fun listBuilding(lore: List<String>): MutableList<Renderable> {
+        val updateList: MutableList<Renderable> = mutableListOf(
+            Renderable.string("§d§lTaming 60 Cost: §r§d(${
+                if (config.otherRarities) "cheapest" else "exact"
+            } rarity)")
+        )
+        var totalCost = 0.0
+        for (line in lore) {
+            neededPetPattern.matchMatcher(line) {
+                val origTierString = group("tier")
+                val origPetString = group("pet")
+                val pet = origPetString.uppercase().replace(" ", "_").removePrefix("FROST_")
+                val originalTier = LorenzRarity.entries.find { it.name == origTierString.uppercase() }!!.id
+                val (petPrice, cheapestTier) = findCheapestTier(pet, originalTier)
+                val displayPetString =
+                    if (cheapestTier == originalTier) group("fullThing")
+                    else "${LorenzRarity.entries.find { it.id == cheapestTier }!!.formattedName} $origPetString"
+                if (petPrice != -1.0) {
+                    totalCost += petPrice
+                    updateList.add(Renderable.clickAndHover(
+                        text = " §7- $displayPetString§7: §6${petPrice.addSeparators()} coins",
+                        tips = listOf(
+                            "§aClick to run §e/ahs ] $origPetString §ato find it on the Auction House.",
+                            "§aNotes: §eSet the rarity filter yourself. §cBooster Cookie required!"
+                        ),
+                        onClick = { LorenzUtils.sendCommandToServer("ahs ] $origPetString") }
+                    ))
+                } else {
+                    updateList.add(Renderable.clickAndHover(
+                        text = " §7- $displayPetString§7: §eNot on AH; view its Wiki article here.",
+                        tips = listOf("§4Click to run §e/wiki $pet §4to view how to obtain it."),
+                        onClick = { LorenzUtils.sendCommandToServer("wiki $pet") }
+                    ))
+                }
+            }
+        }
+        updateList.add(Renderable.string("§dTotal cost §7(§6Lowest BIN§7): §6${totalCost.addSeparators()} coins"))
+        if (config.otherRarities) updateList.add(Renderable.string("§c§lDisclaimer:§r§c Total does not include costs to upgrade via Kat."))
+        return updateList
+    }
+
+    private fun findCheapestTier(pet: String, originalTier: Int): Pair<Double, Int> {
+        val petPriceOne = "$pet$SEPARATOR$originalTier".getPetPrice()
```

Comment:
"$pet$SEPARATOR$originalTier" please extract that as a function. Those strings are really hard to read, and at 3 places.